ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -683,6 +708,7 @@ private void processGetBlobResponse(GetRequestInfo getRequestInfo, GetResponse g
               // this is a successful response and one that completes the operation regardless of whether the
               // success target has been reached or not.
               chunkCompleted = true;
+              retainExceptionOnSuccess = true;
```

Comment:
`retainExceptionOnSuccess` shouldn't be strictly required at this point, because `checkAndMaybeComplete()` will be called immediately afterward and there shouldn't be anything else that marks the operation as a success, in which case the exception could possibly be overwritten. I added it to communicate that the blob deleted/expired error has the highest priority.